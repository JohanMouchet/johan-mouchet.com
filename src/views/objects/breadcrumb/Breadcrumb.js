// @flow

import * as React from "react";
import cx from "classnames";
import "./___.scss";

/**
 * Breadcrumb
 *
 * @param {string} levels.title                 - Breadcrumb level title
 * @param {string} levels.url                   - Breadcrumb title relative url
 * @param {string} [metadata.contextClass]      - Breadcrumb context class
 * @param {array}  [metadata.additionalClasses] - Breadcrumb additional classes
 */

// TODO: "context" was passed in the macro

const ___ = (props: Props) => {
	const {
	  prop1,
	  prop2
	} = props;
  
	const ___Class = cx(
	  "o-breadcrumb",
	  props.metadata.contextClass,
	  props.metadata.additionalClasses,
	  {
		prop1: "___--prop1"
	  }
	);
  
	return (
	<ol className="___Class">
		{% for level in props.levels %}
			<li className="{ 'o-breadcrumb__level'|contextClass(contextClass) } wow fadeInUp" {% if loop.index > 1 %}data-wow-delay="{ loop.index0 * 0.15 }s"{% endif %}>
				<a className="{ 'o-breadcrumb__link'|contextClass(contextClass) }" {% if level.url and not loop.last %}href="{ level.url }"{% endif %}>{ level.title }</a>
			</li>
		{% endfor %}
	</ol>

	/* <pre> */
	<script type="application/ld+json">
		{
			"@context": "http://schema.org",
			"@type": "BreadcrumbList",
			"itemListElement": [
			{%- for level in props.levels -%}
				{
					"@type": "ListItem",
					"position": { loop.index },
					"item": {
					"@type": "Thing",
						"@id": "{ context._SITE['hosturl'] }{ context._SITE['baseurl'] }{ loop.last ? context._SITE['rurl'] : level.url  }",
						"name": "{ level.title }"
					}
				}
				{- not loop.last ? ',' -}
			{%- endfor -%}
			]
		}
	</script>
	/* </pre> */
  );
};

___.defaultProps = {
  prop1: 2
};

export default ___;
