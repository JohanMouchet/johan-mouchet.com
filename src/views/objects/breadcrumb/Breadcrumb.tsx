import * as React from "react";
import cx from "classnames";
import "./Breadcrumb.scss";

type Props = {
  levels: Array<{
    title: string;
    url: string;
  }>;
  className?: string | string[] | { [key: string]: boolean };
};

const Breadcrumb: React.FC<Props> = ({ levels, className }) => {
  const classNames = cx("o-breadcrumb", className);
  const lastLevel = levels?.length - 1;

  return !levels ? null : (
    <>
      <ol className={classNames}>
        {levels.map((level, index) => (
          <li
            className="o-breadcrumb__level u-animation u-animation--fade-in-up"
            style={{
              animationDelay: index > 0 ? index * 0.15 + "s" : undefined,
            }}
            key={level.url}
          >
            <a
              className="o-breadcrumb__link"
              href={index !== lastLevel ? level.url : undefined}
            >
              {level.title}
            </a>
          </li>
        ))}
      </ol>
      {/* TODO: check if work? */}
      {/* <pre> */}
      {/* <script type="application/ld+json">
       {`
      "@context": "http://schema.org",
      "@type": "BreadcrumbList",
      "itemListElement": [
      ${levels.map(
         (level, index) =>
           ({
             "@type": "ListItem",
             position: index,
             item: {
               "@type": "Thing",
               "@id": level.url,
               name: level.title
             }
           }(index !== lastLevel) && ",")
       )}
      ]
      `}
      </script> */}
      {/* </pre> */}
    </>
  );
};

export default Breadcrumb;
