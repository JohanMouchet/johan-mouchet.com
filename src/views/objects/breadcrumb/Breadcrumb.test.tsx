import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import { Breadcrumb } from "./Breadcrumb";

it("Renders <Breadcrumb /> without crashing", () => {
  expect(
    render(
      <Breadcrumb
        levels={[
          {
            title: "Title",
            link: {
              url: "#",
            },
          },
        ]}
      />
    ).container.firstChild
  ).toMatchSnapshot();
});
