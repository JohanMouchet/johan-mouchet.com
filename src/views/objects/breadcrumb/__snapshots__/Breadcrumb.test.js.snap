// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Breadcrumb /> without crashing 1`] = `""`;
